src/main/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -378,6 +385,10 @@ protected int doWork() {
                 }
             }
 
+            if (DUPLEX_UMI) {
+                rec.setAttribute(MOLECULAR_IDENTIFIER_TAG, UmiUtil.molecularIdentifierString(rec, MOLECULAR_IDENTIFIER_TAG));
```

Comment:
I realize this should probably also be set in all cases where a UMI is used.
I made that change, and now use the UmiUtil method to set the identifier.  The behavior will depend on whether or not DUPLEX_UMI is true/false.

I also set MOLECULAR_IDENTIFIER_TAG as an option (default = null) because in most cases people won't want this polluting their bam file, and it could significantly increase the file size.